packages/components/pie-button/src/defs.ts

```diff
@@ -1,58 +1,11 @@
+export const sizes = ['xsmall', 'small-expressive', 'small-productive', 'medium', 'large'] as const;
+export const types = ['submit', 'button', 'reset', 'menu'] as const;
+export const variants = ['primary', 'secondary', 'outline', 'ghost'] as const;
+
 export interface ButtonProps {
-    /**
```

Comment:
I kind of think we should include JSDoc comments like this as standard in these `defs`.

Completely understand that these are fairly self explanatory examples for these props, but we will have more complex props that these comments will become more useful for.

I think they either should live here, or in the `index.ts` file where they are used (happy for them to live inside the actual component potentially).

Reply:
I think we should use them for the less obvious ones, but having comments like `the size of the button.` isn't particularly helpful in my opinion.

Reply:
It's more a consistency thing – it's better to have them on all props/definitions than only on a couple.

For example, the difference between type and variant may be completely obvious to us, as we work with this code a lot, but to someone new to the code, the extra comments clear it up straight away.

Reply:
I partially agree, however having a JSDoc comment that only says "the variant of the button" doesn't exactly clear up much confusion 😄

Reply:
Exactly :D

Reply:
@ashleynolan can I raise a separate PR to sort out the comments? I see no benefit in leaving them as they were as they weren't very helpful. I could add more descriptive ones in a dedicated PR if that's ok

Reply:
Yes please 👍